packages/dashboard/src/features/database/helpers.ts

```
@@ -1,11 +1,133 @@
+
+/**
+ * Decodes a grid row key produced by {@link encodeRecordKey} back into the
+ * primary-key tuple to send to the record update/delete APIs.
+ */
+function isRecordPrimaryKey(value: unknown): value is RecordPrimaryKey {
+  if (!value || typeof value !== 'object' || Array.isArray(value)) {
+    return false;
+  }
+  return Object.values(value).every(
+    (item) =>
+      typeof item === 'string' || typeof item === 'number' || typeof item === 'boolean'
```

ESLint

[error] Delete `⏎·····`

(prettier/prettier)